Sources/KnitMacrosImplementations/ResolvableMacro.swift

```diff
@@ -0,0 +1,236 @@
+            guard let ret = funcDecl.signature.returnClause?.type.as(IdentifierTypeSyntax.self)?.name.text else {
+                throw DiagnosticsError(
+                    diagnostics: [.init(node: funcDecl, message:  Error.missingReturnType)]
+                )
+            }
+            let isStatic = funcDecl.modifiers.contains { $0.name.text == "static" }
+            guard isStatic else {
+                throw DiagnosticsError(
+                    diagnostics: [.init(node: funcDecl, message:  Error.unsupportAttachment)]
+                )
+            }
+            returnType = ret
+            makeCall = funcDecl.name.text
+        } else {
+            throw DiagnosticsError(
+                diagnostics: [.init(node: node, message:  Error.unsupportAttachment)]
+            )
+        }
+
+        let params = try extractParams(parameterClause: parameterClause)
+
+        return [
+            createMakeFunction(
+                params: params,
+                resolverType: resolverType,
+                makeCall: makeCall,
+                returnType: returnType
+            )
+        ]
+    }
+
+    private static func createMakeFunction(
+        params: [Param],
+        resolverType: String,
+        makeCall: String,
+        returnType: String
+    ) -> DeclSyntax {
+        let paramsResolved = params.map { param in
+            return param.resolveCall
+        }
+        let paramsString = paramsResolved.joined(separator: ",\n")
+        var makeArguments = ["resolver: \(resolverType)"]
+        for param in params {
+            if param.isArgument {
+                makeArguments.append("\(param.name): \(param.type.name)")
+            }
+        }
+
+        let makeArgumentsString = makeArguments.joined(separator: ", ")
+
+       return """
+       static func make(\(raw: makeArgumentsString)) -> \(raw: returnType) {
+            return \(raw: makeCall)(
+                \(raw: paramsString)
+            )
+       }
+       """
+    }
+
+    private static func extractType(typeSyntax: TypeSyntax) throws -> TypeInformation {
+        if let type = typeSyntax.as(IdentifierTypeSyntax.self) {
+            return TypeInformation(name: type.name.text)
+        } else if let type = typeSyntax.as(AttributedTypeSyntax.self) {
+            let baseType = try extractType(typeSyntax: type.baseType)
+            return TypeInformation(name: baseType.name)
+        } else if let type = typeSyntax.as(FunctionTypeSyntax.self) {
+            return TypeInformation(name: "(\(type.description))")
+        }
+        throw DiagnosticsError(
+            diagnostics: [.init(node: typeSyntax, message:  Error.invalidParamType(typeSyntax.description))]
+        )
+    }
+
+    // Identify any property wrappers that change how types are resolved
+    private static func extractHint(paramSyntax: FunctionParameterSyntax) -> ParamHint? {
+        for element in paramSyntax.attributes {
```

Comment:
This did make me think if we should provide an error if someone uses `@Argument @Named("foo")` as those conflict. Not blocking